runtime/compiler/control/JITServerHelpers.cpp

```diff
@@ -251,6 +449,29 @@ packArrayROMClassData(const J9ROMClass *romClass, ROMClassPackContext &ctx)
       }
    }
 
+// Calculate the size of the portion of a ROM class that occurs before its UTF8 string section (the pre-string section),
+// without any padding that may have been added during initial ROM class writing.
+//
+// Why is this necessary? The sections inside a ROMClass before the UTF8 string section are 32-bit aligned, but the
+// entire pre-string section needs to be 64-bit aligned. That means exactly four bytes of padding may have been added
+// to the end of the pre-string section during initial ROM class building. When we strip out the debug
+// info from a ROM class, we might reintroduce a misalignment in this pre-string section. Two scenarios are possible:
+//
+// 1. The original ROM class was padded, in which case the solution is to drop that four bytes of padding
+// 2. The original ROM class was not padded, in which case the solution is to introduce four bytes of padding
+//
+// We can only distinguish between scenarios (1) and (2) if we know the unpadded size of the pre-string section.
+static void
+sectionEndCallback(J9ROMClass *romClass, void *sectionPtr, uintptr_t sectionSize, const char *sectionName, void *userData)
+   {
+   auto ctx = (ROMClassPackContext *)userData;
+   if (ctx->isInline(sectionPtr, romClass))
+      {
+      size_t endOffset = ((uint8_t *)sectionPtr - (uint8_t *)romClass) + sectionSize;
+      ctx->_preStringSize = std::max(ctx->_preStringSize, endOffset);
```

Comment:
It doesn't ensure alignment. We need to get the unaligned (unpadded) size of the ROM class before the string section, so that when we calculate the aligned (padded) size of the packed version of the ROM class here:

https://github.com/eclipse-openj9/openj9/pull/18848/files#diff-d88e87e324a1ed193ea9af7d85837cc118c61bdca9ae8380719b945bbffd0757R611

we don't introduce unnecessary padding.

I'll clarify that in the comment, and I should probably add that 

1. The "pre-string section" isn't a real section that the ROM class walk visits (neither is the UTF8 string section)
2. The ROM class walker visits sections that are either entirely within the pre-string section or are entirely outside the ROM class. In particular, it doesn't visit anything in the UTF8 string section or the intermediate class data section.
3. The ROM class walker visits every section before the UTF8 string section.

That's why taking the max actually gives you the size of the pre-string section, and why going through the trouble of taking the max is necessary in the first place. (Maybe "pre-string area" is a better term for it).

The assert here:

https://github.com/eclipse-openj9/openj9/pull/18848/files#diff-d88e87e324a1ed193ea9af7d85837cc118c61bdca9ae8380719b945bbffd0757R605

should hopefully catch any case where someone changes the ROM class format or walker significantly enough that those properties are violated.